sklearn/feature_selection/_mutual_info.py

```diff
@@ -301,10 +303,12 @@ def _estimate_mi(
             * rng.standard_normal(size=n_samples)
         )
 
-    mi = [
-        _compute_mi(x, y, discrete_feature, discrete_target, n_neighbors)
+    mi = Parallel(n_jobs=n_jobs)(
+        delayed(lambda x, df: _compute_mi(x, y, df, discrete_target, n_neighbors))(
+            x, discrete_feature
+        )
         for x, discrete_feature in zip(_iterate_columns(X), discrete_mask)
-    ]
+    )
```

Comment:
I might be missing something here, but you not just passing the variable instead of doing a "partial" function call?

```suggestion
    mi = Parallel(n_jobs=n_jobs)(
        delayed(_compute_mi)(x, y, discrete_feature, discrete_target, n_neighbors)
        for x, discrete_feature in zip(_iterate_columns(X), discrete_mask)
    )
```